tsdb/tombstones/tombstones.go

```diff
@@ -145,6 +138,44 @@ func WriteFile(logger log.Logger, dir string, tr Reader) (int64, error) {
 	return int64(size), fileutil.Replace(tmp, path)
 }
 
+func Encode(tr Reader) ([]byte, error) {
+	buf := encoding.Encbuf{}
+	buf.PutByte(tombstoneFormatV1)
+	err := tr.Iter(func(ref uint64, ivs Intervals) error {
+		for _, iv := range ivs {
+			buf.PutUvarint64(ref)
+			buf.PutVarint64(iv.Mint)
+			buf.PutVarint64(iv.Maxt)
+		}
+		return nil
+	})
+	return buf.Get(), errors.Wrap(err, "encoding tombstones")
```

Comment:
```suggestion
	return buf.Get(),  tr.Iter(func(ref uint64, ivs Intervals) error {
		for _, iv := range ivs {
			buf.PutUvarint64(ref)
			buf.PutVarint64(iv.Mint)
			buf.PutVarint64(iv.Maxt)
		}
		return nil
	})
```

Reply:
Ah, tried this and it wont work :). Because `buf.Get()` gets evaluated before it can be filled in `tr.Iter`. Moving back to the old one but without error wrapping.